src/window.rs

```diff
@@ -0,0 +1,177 @@
+use core::marker::PhantomData;
+use core::f64;
+
+use {Sample, FloatSample};
+use signal;
+use signal::{Phase, ConstHz};
+use conv::{FromSample, ToSample};
+use frame::Frame;
+
+// Using types instead of enum allows for static dispatch
+pub trait Type<S: Sample> {
+    fn at_phase(phase: S) -> S;
+}
+
+pub struct Hanning;
+
+pub struct Rectangle;
+
+pub struct Window<F, WT> 
+    where F: Frame,
+          WT: Type<F::Sample>
+{
+    pub phase: Phase<ConstHz>,
+    ftype: PhantomData<F>,
+    wttype: PhantomData<WT>
+}
+
+pub struct WindowedFrame<'a, F, WT> 
+    where F: 'a + Frame,
+          F::Sample: FromSample<f64>, 
+          WT: Type<F::Sample>
+{
+    pub data: &'a [F],
+    pub window: Window<F, WT>,
+    idx: usize
+}
+
+/// Windower takes a long slice of data and generates an iterator over its frames
+pub struct Windower<'a, F, WT> 
+    where F: 'a + Frame, 
+          F::Sample: FromSample<f64>, 
+          WT: Type<F::Sample>
+{
+    pub bin: usize,
+    pub hop: usize,
+    pub idx: usize,
+    pub data: &'a [F],
+    wttype: PhantomData<WT>
+}
+
+impl<S: Sample + ToSample<f64> + FromSample<f64>> Type<S> for Hanning {
+    fn at_phase(phase: S) -> S {
+        let v = phase.to_sample::<f64>() * f64::consts::PI * 2.;
```

Comment:
I'm making some small follow up changes atm and realised that perhaps `Sample` isn't the best trait for the phase? For example, if the user is working with a `Window` of `i16` samples, the phase in turn would also be type `i16` which I don't think would give the correct result at this line.

How about for now, we just assume that the phase must be `f64` until we can come up with a more generic trait for `Phase`? I think I'd like to try and get #26 in first before working on such a trait though.